### PR TITLE
New json tests

### PR DESCRIPTION
**Pull request by justinjmoses**
- Created a new test target `test:deployments` which tests the following
    - all synths in synths.json are added to `Synthetix`, and inverses are set as expected
    - all `AddressResolver` addresses are set
    - all ABIs from Etherscan are set (verified) and match expected ones
- Also fixed the testnet deployment JSON files where `ProxysUSD` had source incorrectly set to `Proxy` not `ProxyERC20`
- And reset the inverse pricing on the inverted synths in all testnets to match mainnet

cc @clementbalestrat - this will replace all those tests in synthetix-js checking contract addresses that don't belong there

### CODE REVIEW

File: migrations/2_deploy_synthetix_system.js [hunk 1]
```diff
@@ -450,7 +449,6 @@ module.exports = async function(deployer, network, accounts) {
 			feePoolEternalStorage.address,
 			feePoolState.address,
 			issuer.address,
-			etherCollateral.address, // MultiCollateral for Synth uses EtherCollateral
```
Comment (jacko125) [inline migrations/2_deploy_synthetix_system.js]: @justinjmoses  Is the idea that etherCollateral won't be updated any time soon?

Comment (hav-noms) [reply migrations/2_deploy_synthetix_system.js]: It's in the list up higher as EtherCollateral not MultiCollateral but yeah sETH needs this address in the resolver

Comment (justinjmoses) [reply migrations/2_deploy_synthetix_system.js]: Recall that SynthsETH is given `EtherCollateral` as a constructor argument. This was leftover.

Comment (jacko125) [reply migrations/2_deploy_synthetix_system.js]: Correct, so no other contracts directly looks up `MultiCollateral` key https://github.com/Synthetixio/synthetix/blob/master/contracts/MultiCollateralSynth.sol#L27-L40

Comment (justinjmoses) [reply migrations/2_deploy_synthetix_system.js]: Correct, so `MultiCollateral` is leftover that needs to be removed

**Pull request merged**